core/src/functionalTest/groovy/com/novoda/gradle/release/JavaDifferentGradleVersions.groovy

```diff
@@ -0,0 +1,103 @@
+package com.novoda.gradle.release
+
+import org.assertj.core.api.Assertions
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.*
+import org.junit.rules.TemporaryFolder
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(JUnit4.class)
+class JavaDifferentGradleVersions {
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder()
+
+    private GradleRunner runner
+
+    @Before
+    void setUp() throws Exception {
+        runner = GradleRunner.create()
+                .withProjectDir(temporaryFolder.root)
+                .withArguments("build", "bintrayUpload", "-PbintrayKey=key", "-PbintrayUser=user")
+                .withPluginClasspath()
+
+        def buildFile = temporaryFolder.newFile("build.gradle")
+        buildFile.write(
+                """            
+            plugins { 
+                id 'java-library'
+                id 'com.novoda.bintray-release' version '0.8.0' 
+            }
+            
+            repositories {
+                jcenter()
+            }
+            
+            dependencies {
+                implementation "junit:junit:4.12"
+            }
+            
+            publish {
+                userOrg = 'novoda'
+                groupId = 'com.novoda'
+                artifactId = 'test'
+                publishVersion = '1.0'
+            }
+                    """
+        )
+
+        def fakeSourceCode = new File(temporaryFolder.root, "src/main/java/HelloWorld.java")
```

Comment:
Same comment here. 
Also please have a look at [TestProjectRule](https://github.com/novoda/gradle-static-analysis-plugin/blob/master/plugin/src/test/groovy/com/novoda/test/TestProjectRule.groovy). It basically makes it easier to create such projects in tests.